x/distribution/keeper/grpc_query.go

```diff
@@ -269,22 +269,26 @@ func (k Querier) DelegationTotalRewards(ctx context.Context, req *types.QueryDel
 		func(_ int64, del sdk.DelegationI) (stop bool) {
 			valAddr, err := k.stakingKeeper.ValidatorAddressCodec().StringToBytes(del.GetValidatorAddr())
 			if err != nil {
-				panic(err)
+				k.Logger(ctx).Error(err.Error())
```

Comment:
My issue is still that the error is discarded if we simply log it.
Cannot we store it somewhere instead and return it? The query will give inaccurate delegation rewards otherwise and silently log an error to the node.